src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java

```diff
@@ -0,0 +1,126 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.IColony;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.minecolonies.coremod.commands.AbstractSingleCommand.Commands.HOMETP;
+
+/**
+ * this command is made to TP a player to a safe random spot that is not to close to another colony.
+ * Need to add a configs permissions check.
+ * Need to allow OPs to send players ./mc ctp (Player) if player is not allowed.
+ */
+public class HomeTeleportCommand extends AbstractSingleCommand
+{
+    public static final  String DESC             = "home";
+    private static final String CANT_FIND_COLONY = "No Colony found for teleport, please define one.";
+    private static final String CANT_FIND_PLAYER = "No player found for teleport, please define one.";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public HomeTeleportCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "home";
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+
```

Comment:
every time I do it moves the curly bracket up to the line and then sonar freaks out because sonar is egotistical and thinks everything should be his way....